modules/text-input/react/lib/TextInput.tsx

```diff
@@ -9,18 +9,11 @@ import InputIconContainer from './InputIconContainer';
 export interface TextInputProps
   extends GrowthBehavior,
     React.InputHTMLAttributes<HTMLInputElement> {
-  disabled?: boolean;
   error?: ErrorType;
-  hasIcon?: boolean;
-  inputRef?: React.Ref<HTMLInputElement>;
-  onChange?: React.ChangeEventHandler<HTMLInputElement>;
-  placeholder?: string;
-  readOnly?: boolean;
-  type?: string;
-  value?: any;
+  inputRef?: React.RefObject<HTMLInputElement>;
```

Comment:
since these were all normal props that are just being spread on the input we don't need to call them out. Also looks like hasIcon was only included for use in the type below. It's value is computed using the error prop.

Reply:
It's been conventional for us to explicitly call out these normal props that are used for functionality within the component (we do the same on checkbox, radio, etc. as well). If we want to change this, I don't think it should happen in this PR

Good catch on `hasIcon` 👍

Reply:
are they used for functionality? Looks like they are just getting spread into inputProps